CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(shared_ptr)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
+add_definitions("-std=c++17")
+
+if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
```

Comment:
is conan necessary?

Reply:
Of course not. But it is handy way to build project and link library out of the box without explicit installation third parties.

Reply:
Separate installation is known practice, generally used inside linux ecosystem and currently starts to be supported on microsoft as well. Please feel free to keep your GTest delivery mechanism if it is comfortable for you